packages/website-backend/src/services/real-time/MutationEventServerOrchestrator.ts

```diff
@@ -0,0 +1,41 @@
+import { createServer } from 'http';
+import { MutationEventServer } from './MutationEventServer.js';
+import { SseServer } from './SseServer.js';
+import { Service } from '@tsed/di';
+import { ReportIdentifier } from '@stryker-mutator/dashboard-common';
+import Configuration from '../Configuration.js';
+
+@Service()
+export default class MutationEventServerOrchestrator {
+  #mutationEventServers = new Map<string, MutationEventServer>();
+  #configuration: Configuration;
+
+  constructor(configuration: Configuration) {
+    this.#configuration = configuration;
+  }
+
+  getSseInstanceForProject(identifier: ReportIdentifier): MutationEventServer {
+    const id = this.#toId(identifier);
+    if (!this.#mutationEventServers.has(id)) {
+      return this.#createMutationEventServer(id);
+    }
+
+    return this.#mutationEventServers.get(id)!;
+  }
+
+  #toId(identifier: ReportIdentifier) {
+    return Object.values(identifier).join(';');
+  }
+
+  #createMutationEventServer(id: string): MutationEventServer {
+    const server = new MutationEventServer(
+      new SseServer(createServer(), this.#configuration.cors)
```

Comment:
Why do we create a new server instead of reusing the existing one? It seems straightforward to inject the server itself: https://tsed.io/docs/factory.html#inject-http-server-or-https-server

Reply:
Correct, we do make a new server. However we do not start the server (see the `start()` method), instead we 'attach' to the incoming request (see the `attach(...)`) method.

So everytime a `/sse` request comes in, it attaches itself to that request. I think the reason this was done, is because I reused a lot of functionality from the `SseServer` that is in `mutation-testing-elements`.

Reply:
Maybe the `handleRequest` method should become a static one? Might make more sense to do that instead.

Reply:
I think this should just be a map with the slug as the key and a list of connections as the value.